src/main/java/nl/tudelft/lifetiles/graph/controller/GraphController.java

```diff
@@ -213,45 +253,88 @@ private void repaint() {
             }
             view = new TileView(this);
 
-            root = new Group();
-
             wrapper.hvalueProperty().addListener(
                     (observable, oldValue, newValue) -> {
-                        repaintPosition(root, newValue.doubleValue());
+                        repaintPosition(newValue.doubleValue());
                     });
 
-            Rectangle clip = new Rectangle(getMaxUnifiedEnd(graph)
-                    * VertexView.HORIZONTALSCALE, 0);
-            root.getChildren().add(clip);
-
-            repaintPosition(root, wrapper.hvalueProperty().doubleValue());
+            repaintPosition(wrapper.hvalueProperty().doubleValue());
         }
     }
 
     /**
      * Repaints the view indicated by the bucket in the given position.
      *
-     * @param root
-     *            Root group used to store the visualized graph in.
      * @param position
      *            Position in the scrollPane.
      */
-    private void repaintPosition(final Group root, final double position) {
-        int nextPosition = getBucketPosition(position);
-        if (currentPosition != nextPosition || forceRepaintPosition) {
+    private void repaintPosition(final double position) {
```

Comment:
This method is too long, you should split it up, or even better, move it to a different class so you can test it.